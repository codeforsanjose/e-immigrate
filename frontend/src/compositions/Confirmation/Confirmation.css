.Confirmation {
    max-width: 600px;
    margin: auto;
    display: grid;
}

.Confirmation h1 {
    font-size: 1.5rem;
    font-weight: 700;
    margin: 0 0 1rem 0;
}

.Confirmation h2 {
    font-size: 20px;
    font-weight: 600;
    margin: 0 0 10px 0;
}

.Confirmation p {
    font-size: 17px;
    font-weight: 400;
    margin: 0 0 0.5rem 0;
}

.Confirmation div {
    margin: 0 0 54px 0;
}

.Confirmation ul {
    list-style-image: url('../../data/images/CIIT_bullet.svg');
}

.Confirmation li {
    margin-bottom: 1rem;
}
