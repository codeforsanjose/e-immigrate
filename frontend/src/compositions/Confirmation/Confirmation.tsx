import React from 'react';

import './Confirmation.css';
import { useContentContext } from '../../contexts/ContentContext';


export function Confirmation() {
    const { content } = useContentContext();
    return (
        <div className="Confirmation">
            <div>
                <h1>
                    {content.step3Header}: {content.step3Title}
                </h1>
                <h2>{content.step3Header2}</h2>
                <p>{content.step3Text2}</p>
            </div>
            <div>
                <h1>{content.step3Header3}</h1>
                <h2>{content.step3Text3}</h2>
                <ul>
                    <li>{content.step3Tip1}</li>
                    <li>{content.step3Tip2}</li>
                    <li>{content.step3Tip3}</li>
                </ul>
            </div>
            <div>
                <h2>{content.step3Text4}</h2>
            </div>
            <p>{content.step3Text5}</p>
            <p>
                {content.step3Text6}{' '}
                <a href="https://www.e-immigrate.info/" target="_blank" rel="noreferrer">
                    e-immigrate.info
                </a>
            </p>
        </div>
    );
}
