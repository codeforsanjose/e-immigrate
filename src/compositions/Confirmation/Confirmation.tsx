import React from 'react';

import './Confirmation.css';

type ConfirmationProps = {
    content: {
        step3Header: string;
        step3Title: string;
        step3Header2: string;
        step3Text2: string;
        step3Header3: string;
        step3Text3: string;
        step3Tip1: string;
        step3Tip2: string;
        step3Tip3: string;
        step3Text4: string;
        step3Text5: string;
        step3Text6: string;
    };
};
export function Confirmation(props: ConfirmationProps) {
    const {
        content,
    } = props;
    return (
        <div className="Confirmation">
            <div>
                <h1>
                    {content.step3Header}: {content.step3Title}
                </h1>
                <h2>{content.step3Header2}</h2>
                <p>{content.step3Text2}</p>
            </div>
            <div>
                <h1>{content.step3Header3}</h1>
                <h2>{content.step3Text3}</h2>
                <ul>
                    <li>{content.step3Tip1}</li>
                    <li>{content.step3Tip2}</li>
                    <li>{content.step3Tip3}</li>
                </ul>
            </div>
            <div>
                <h2>{content.step3Text4}</h2>
            </div>
            <p>{content.step3Text5}</p>
            <p>
                {content.step3Text6}{' '}
                <a href="https://www.e-immigrate.info/" target="_blank">
                    e-immigrate.info
                </a>
            </p>
        </div>
    );
}
