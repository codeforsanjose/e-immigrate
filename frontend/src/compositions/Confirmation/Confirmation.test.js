import React from 'react';
import Confirmation from './Confirmation';

import renderer from 'react-test-renderer';

const content = {
    step3Header: 'Step 3',
    step3Header2: 'Congratulations, you have completed the questionnaire!',
    step3Header3: "What's Next?",
    step3Instructions:
        'Overview of next steps to receive free legal assistance',
    step3Text2:
        'We have sent a copy of this confirmation page to the email and/or phone that you provided in the questionnaire.',
    step3Text3:
        'If it appears that you are eligible for citizenship based on the answers you provided:',
    step3Text4:
        'If you are ineligible to apply for citizenship at this time we will follow up with you by email no later than Monday, May 24th.',
    step3Text5:
        'Note: Due to limited space and COVID-19, we cannot guarantee that you will receive services. Thank you for your cooperation, patience and understanding.',
    step3Text6: 'If you want more information about immigration, please visit',
    step3Tip1:
        'Project New Citizen team members will begin to contact all eligible participants starting Monday, May 17th for a FREE follow-up consultation to verify all the information, give advice, and instuction.',
    step3Tip2:
        'If you are eligible to apply, you will be assisted by the immigration attourneys and DOJ accredited representatives. The citizenship application assisstance will be FREE of charge, and all information will be kept confidential.',
    step3Tip3:
        'You will also receive resources to help you study for the citizenship interview.',
    step3Title: 'Confirmation',
};

it('confirmation page renders correctly', () => {
    const tree = renderer.create(<Confirmation content={content} />).toJSON();
    expect(tree).toMatchSnapshot();
});
