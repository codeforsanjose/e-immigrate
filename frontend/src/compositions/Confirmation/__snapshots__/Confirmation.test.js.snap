// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirmation page renders correctly 1`] = `
<div
  className="Confirmation"
>
  <div>
    <h1>
      Step 3
      : 
      Confirmation
    </h1>
    <h2>
      Congratulations, you have completed the questionnaire!
    </h2>
    <p>
      We have sent a copy of this confirmation page to the email and/or phone that you provided in the questionnaire.
    </p>
  </div>
  <div>
    <h1>
      What's Next?
    </h1>
    <h2>
      If it appears that you are eligible for citizenship based on the answers you provided:
    </h2>
    <ul>
      <li>
        Project New Citizen team members will begin to contact all eligible participants starting Monday, May 17th for a FREE follow-up consultation to verify all the information, give advice, and instuction.
      </li>
      <li>
        If you are eligible to apply, you will be assisted by the immigration attourneys and DOJ accredited representatives. The citizenship application assisstance will be FREE of charge, and all information will be kept confidential.
      </li>
      <li>
        You will also receive resources to help you study for the citizenship interview.
      </li>
    </ul>
  </div>
  <div>
    <h2>
      If you are ineligible to apply for citizenship at this time we will follow up with you by email no later than Monday, May 24th.
    </h2>
  </div>
  <p>
    Note: Due to limited space and COVID-19, we cannot guarantee that you will receive services. Thank you for your cooperation, patience and understanding.
  </p>
  <p>
    If you want more information about immigration, please visit
     
    <a
      href="https://www.e-immigrate.info/"
      target="_blank"
    >
      e-immigrate.info
    </a>
  </p>
</div>
`;
